org-gcal.el

```diff
@@ -1384,92 +1373,21 @@ delete calendar info from events on calendars you no longer have access to."
                 (deferred:succeed nil))))
         (deferred:succeed nil)))))
 
-(defun org-gcal-request-authorization ()
-  "Request OAuth authorization at AUTH-URL by launching `browse-url'.
-  CLIENT-ID is the client id provided by the provider.
-  It returns the code provided by the service."
-  (let* ((gcal-auth-url
-          (concat org-gcal-auth-url
-                  "?client_id=" (url-hexify-string org-gcal-client-id)
-                  "&response_type=code"
-                  "&redirect_uri=" (url-hexify-string "urn:ietf:wg:oauth:2.0:oob")
-                  "&scope=" (url-hexify-string org-gcal-resource-url)))
-         (prompt
-          (format
-           "Please visit (if it doesn't open automatically): %s\n\nEnter the code your browser displayed:"
-           gcal-auth-url)))
-    (browse-url gcal-auth-url)
-    (read-string prompt)))
-
-(defun org-gcal-request-token ()
-  "Refresh OAuth access at TOKEN-URL.
-
-  Returns a ‘deferred’ object that can be used to wait for completion."
-  (interactive)
-  (deferred:$
-    (request-deferred
-     org-gcal-token-url
-     :type "POST"
-     :data `(("client_id" . ,org-gcal-client-id)
-             ("client_secret" . ,org-gcal-client-secret)
-             ("code" . ,(org-gcal-request-authorization))
-             ("redirect_uri" .  "urn:ietf:wg:oauth:2.0:oob")
-             ("grant_type" . "authorization_code"))
-     :parser 'org-gcal--json-read)
-    (deferred:nextc it
-      (lambda (response)
-        (let
-            ((data (request-response-data response))
-             (status-code (request-response-status-code response))
-             (error-thrown (request-response-error-thrown response)))
-          (cond
-           ;; If there is no network connectivity, the response will not
-           ;; include a status code.
-           ((eq status-code nil)
-            (org-gcal--notify
-             "Got Error"
-             "Could not contact remote service. Please check your network connectivity.")
-            (error "Network connectivity issue %s: %s" status-code error-thrown))
-           ;; Generic error-handler meant to provide useful information about
-           ;; failure cases not otherwise explicitly specified.
-           ((not (eq error-thrown nil))
-            (org-gcal--notify
-             (concat "Status code: " (number-to-string status-code))
-             (pp-to-string error-thrown))
-            (error "Got error %S: %S" status-code error-thrown))
-           ;; Fetch was successful.
-           (t
-            (when data
-              (setq org-gcal-token-plist data)
-              (org-gcal--save-sexp data org-gcal-token-file))
-            (deferred:succeed nil))))))))
-
-(defun org-gcal--refresh-token ()
+(defun org-gcal--get-access-token (calendar-id)
+  "Return the access token for CALENDAR-ID."
+  (aio-wait-for
+   (oauth2-auto-access-token calendar-id 'org-gcal)))
```

Comment:
Instead of `calendar-id` it should be the user's email.

Reply:
Note also that the [PR 9](https://github.com/rhaps0dy/emacs-oauth2-auto/pull/9), which I'm going to merge at some point, changes the format to `(oauth2-auto-access-token "user@gmail.com" 'org-gcal '(calendar))`